caikit/core/data_model/base.py

```diff
@@ -82,176 +86,162 @@ def __new__(mcs, name, bases, attrs):
-            #
-            # 1. The "protobufs" module in caikit.core (injected via
-            #   import_protobufs)
-            # 2. The "protobufs" module in the derived library.
-            #
-            # The second option is primarily needed when using import_tracker on
-            # a caikit.core derived library. The side-effects of
-            # import_protobufs break the import_tracker mechanism, so this
-            # fallback avoids the reliance on side-effects.
+        if name not in ["DataBase", "DataObjectBase"]:
+            # Look for a precompiled proto class and if found, parse its
+            # descriptor
             proto_class = attrs.get("_proto_class")
-            if proto_class is None:
-                parent_mod = attrs.get("__module__")
-                log.debug3(
-                    "No proto class found in central registry for [%s]. Looking in [%s]",
-                    name,
-                    parent_mod,
+            if proto_class is not None:
+                fields = tuple(proto_class.DESCRIPTOR.fields_by_name)
+
+            # Otherwise, we need to get the fields from a "special" attribute
+            else:
+                fields = attrs.pop(mcs._FWD_DECL_FIELDS, None)
+                log.debug4(
+                    "Using dataclass forward declaration fields %s for %s", fields, name
                 )
-                if parent_mod is not None:
-                    parent_mod_name = parent_mod.rpartition(".")[0]
-                    if not parent_mod_name:
-                        log.debug3(
-                            "No parent module for data model declared outside library"
-                        )
-                    else:
-                        parent_mod_protos_name = ".".join(
-                            [parent_mod_name, "protobufs"]
-                        )
-                        try:
-                            parent_mod_protos = importlib.import_module(
-                                parent_mod_protos_name
-                            )
-                            proto_class = getattr(parent_mod_protos, name, None)
-                        except ImportError:
-                            log.debug3(
-                                "Could not find a protobufs module in [%s]: %s",
-                                parent_mod,
-                                parent_mod_protos_name,
-                            )
-
-            if proto_class is None:
-                raise AttributeError(
-                    f"Failed to find {name} in caikit.core.data_model.protobufs or derived "
-                    f"library protobufs"
+                error.value_check(
+                    "<COR49310991E>",
+                    fields is not None,
+                    name,
+                    msg="No proto class found for {}",
                 )
 
-            # all fields
-            fields += tuple(proto_class.DESCRIPTOR.fields_by_name)
```

Comment:
All of this logic moved to `parse_proto_descriptor` below so that it could be lazily invoked in `@dataobject`